nbcelltests/tests/test_test.py

```diff
@@ -420,4 +420,75 @@ def test_coverage(self):
             raise ValueError("Cell coverage test should have failed.")
 
 
+######
+
+# should split this file up - but only after deciding on organization
+# of module being tested
+
+# TODO: there's a repeated pattern of cleaning up files generated
+# during tests, but address
+# https://github.com/jpmorganchase/nbcelltests/issues/125 before
+# deciding how to handle that better here.
+
+def test_basic_runWithReturn_pass():
+    """Basic check - just that it runs without error"""
+    generates = os.path.join(os.path.dirname(__file__), "_cell_coverage_test.py")
+    if os.path.exists(generates):
+        raise ValueError("Going to generate %s but it already exists." % generates)
+
+    try:
+        _ = runWithReturn(COVERAGE, rules={'cell_coverage': 10})
+    finally:
+        try:
+            os.remove(generates)
+        except Exception:
+            pass
+
+
+def test_basic_runWithReturn_fail():
+    """Basic check - just that it fails"""
+    generates = os.path.join(os.path.dirname(__file__), "_cell_coverage_test.py")
+    if os.path.exists(generates):
+        raise ValueError("Going to generate %s but it already exists." % generates)
+
+    try:
+        _ = runWithReturn(COVERAGE, rules={'cell_coverage': 100})
+    except Exception:
+        pass  # would need to alter run fn or capture output to check more exactly
+    else:
+        raise ValueError("coverage check should have failed, but didn't")
+    finally:
+        try:
+            os.remove(generates)
+        except Exception:
+            pass
+
+
+def test_basic_runWithReport_pass():
+    """Basic check - just that it runs without error"""
+    generates = os.path.join(os.path.dirname(__file__), "_cell_coverage_test.py")
+    if os.path.exists(generates):
+        raise ValueError("Going to generate %s but it already exists." % generates)
+
+    from nbcelltests.define import TestType
+    try:
+        ret = runWithReport(COVERAGE, executable=None, rules={'cell_coverage': 10})
+    finally:
+        try:
+            os.remove(generates)
+        except Exception:
+            pass
+
+    assert len(ret) == 1
+    assert (ret[0].passed, ret[0].type, ret[0].message) == (1, TestType.CELL_COVERAGE, 'Testing cell coverage')
+
+
+# def test_basic_runWithReport_fail():
+#    from nbcelltests.define import TestType
+#    # TODO it fails here, but it shouldn't, right? we want to be able to report
+#    ret = runWithReport(COVERAGE, executable=None, rules={'cell_coverage':100})
+#    assert len(ret) == 1
+#    assert (ret[0].passed, ret[0].type, ret[0].message) == (False, TestType.CELL_COVERAGE, 'Testing cell coverage')
```

Comment:
Is this future work? I would maybe uncomment it then and either skip or mark as expected failure (non-strict). Dead code should be purged 🔥

Reply:
Yes, I should have linked to https://github.com/jpmorganchase/nbcelltests/issues/126 and https://github.com/jpmorganchase/nbcelltests/issues/122. I need to understand what all the functions are _meant_ to do before going any further.